src/server/game/Globals/ObjectMgr.cpp

```diff
@@ -4015,7 +4015,7 @@ void ObjectMgr::LoadQuests()
 
     // Load `quest_objectives`
     //                                   0   1        2     3             4         5       6      7       8                  9
-    result = WorldDatabase.Query("SELECT ID, QuestID, Type, StorageIndex, ObjectID, Amount, Flags, Flags2, ProgressBarWeight, Description FROM quest_objectives ORDER BY StorageIndex ASC");
+    result = WorldDatabase.Query("SELECT ID, QuestID, Type, StorageIndex, ObjectID, Amount, Flags, Flags2, ProgressBarWeight, Description FROM quest_objectives ORDER BY Order ASC");
```

Comment:
i would suggest to order them by Order and StorageIndex to not break existing data